packages/workspace/src/browser/workspace-variable-contribution.ts

```diff
@@ -55,11 +55,12 @@ export class WorkspaceVariableContribution implements VariableContribution {
                 });
             }
         });
-        for (const widget of this.shell.widgets) {
+        Array.prototype.forEach.call(this.shell.widgets, (widget: Widget) => {
```

Comment:
What? If you did that because `this.shell.widgets` was typed as `readonly any[]`, then you can revert this.

Make sure to run `yarn install` so that `@phosphor/*` types are pulled, and then run the `Restart TS server` command in your IDE to make sure that newly installed/generated typings are available.

Reply:
Thanks for your help @marechal-p!

Interestingly, i only get the following error, when running the tests. The error does not appear in the editor or when building with `yarn install`. The same still happens after running `yarn install` and `Restart TS Server`.

```
  1) "before each" hook for "createOpenWorkspaceOpenFileDialogProps - OS: Linux, Environment: browser, Multi-root workspace: yes":
     Error: @postConstruct error in class WorkspaceVariableContribution: this.shell.widgets is not iterable
      at _postConstruct (/home/simon/Git/theia/node_modules/inversify/lib/resolution/instantiation.js:36:19)
      at Object.resolveInstance (/home/simon/Git/theia/node_modules/inversify/lib/resolution/instantiation.js:53:5)
      at /home/simon/Git/theia/node_modules/inversify/lib/resolution/resolver.js:75:42
      at Array.map (<anonymous>)
      at _injectProperties (/home/simon/Git/theia/node_modules/inversify/lib/resolution/instantiation.js:17:57)
      at Object.resolveInstance (/home/simon/Git/theia/node_modules/inversify/lib/resolution/instantiation.js:48:18)
      at /home/simon/Git/theia/node_modules/inversify/lib/resolution/resolver.js:75:42
      at Object.resolve (/home/simon/Git/theia/node_modules/inversify/lib/resolution/resolver.js:99:12)
      at /home/simon/Git/theia/node_modules/inversify/lib/container/container.js:335:37
      at Container._get (/home/simon/Git/theia/node_modules/inversify/lib/container/container.js:326:44)
      at Container.get (/home/simon/Git/theia/node_modules/inversify/lib/container/container.js:243:21)
      at Context.<anonymous> (lib/browser/workspace-uri-contribution.spec.js:69:31)
      at processImmediate (internal/timers.js:461:21)
´´´

Reply:
Going where the test fails, it is most likely a mistake made when mocking `ShellApplication`.

https://github.com/eclipse-theia/theia/blob/d4642c8182875728c711e1fb1712eec7b5cbf44b/packages/workspace/src/browser/workspace-uri-contribution.spec.ts#L47-L51

Should be `widgets: []`.